core/include/cubos/core/log.hpp

```diff
@@ -353,6 +360,8 @@ namespace cubos::core
         static bool read(std::size_t& cursor, Entry& entry);
 
     private:
+        std::unique_ptr<memory::Stream> mLogFileStream;
```

Comment:
This is a singleton, you can/should just use the global static `State` structure defined in the `log.cpp` file to store this stream.